src/key.rs

```diff
@@ -148,13 +155,26 @@ impl SecretKey {
         }
     }
 
+    /// Converts a `SECRET_KEY_SIZE`-byte slice to a secret key without checking
+    #[inline]
+    pub fn from_slice_no_check(data: &[u8])-> Result<SecretKey, Error> {
+        match data.len() {
+            constants::SECRET_KEY_SIZE => {
+                let mut ret = [0; constants::SECRET_KEY_SIZE];
+                ret[..].copy_from_slice(data);
+                Ok(SecretKey(ret))
+            }
+            _ => Err(InvalidSecretKey)
+        }
+    }
+
     #[inline]
     /// Negates one secret key.
     pub fn negate_assign(
         &mut self
     ) {
         unsafe {
-            let res = ffi::secp256k1_ec_seckey_negate(
+            let res = ffi::secp256k1_ec_privkey_negate(
```

Comment:
These APIs are deprecated upstream. It would be a good idea to update the FFI library to use the latest `seckey` version.

Reply:
Couldn't agree more. For now, I think it's safe to use the deprecated one until someone add the newer function into our secp256k1-zkp.